rls/src/main/java/io/grpc/rls/RlsRequestFactory.java

```diff
@@ -40,29 +41,24 @@
 final class RlsRequestFactory {
 
   private final String target;
-  /**
-   * schema: Path(/serviceName/methodName or /serviceName/*), rls request headerName, header fields.
-   */
-  private final Table<String, String, NameMatcher> keyBuilderTable;
+  private final Map<String, GrpcKeyBuilder> keyBuilderTable;
 
   RlsRequestFactory(RouteLookupConfig rlsConfig, String target) {
     checkNotNull(rlsConfig, "rlsConfig");
     this.target = checkNotNull(target, "target");
     this.keyBuilderTable = createKeyBuilderTable(rlsConfig);
   }
 
-  private static Table<String, String, NameMatcher> createKeyBuilderTable(
+  private static Map<String, GrpcKeyBuilder> createKeyBuilderTable(
       RouteLookupConfig config) {
-    Table<String, String, NameMatcher> table = HashBasedTable.create();
+    Map<String, GrpcKeyBuilder> table = new HashMap<>();
     for (GrpcKeyBuilder grpcKeyBuilder : config.getGrpcKeyBuilders()) {
-      for (NameMatcher nameMatcher : grpcKeyBuilder.getHeaders()) {
-        for (Name name : grpcKeyBuilder.getNames()) {
-          String method =
-              name.getMethod() == null || name.getMethod().isEmpty()
-                  ? "*" : name.getMethod();
-          String path = "/" + name.getService() + "/" + method;
-          table.put(path, nameMatcher.getKey(), nameMatcher);
-        }
+      for (Name name : grpcKeyBuilder.getNames()) {
+        String method =
+            name.getMethod() == null || name.getMethod().isEmpty()
+                ? "*" : name.getMethod();
```

Comment:
nit: I know you're carrying this code from the prev version, but would be cool to move `name.getMethod() == null || name.getMethod().isEmpty()` to a separate variable for readability.

```suggestion
        bool hasMethod = name.getMethod() == null || name.getMethod().isEmpty();
        String method = hasMethod ? "*" : name.getMethod();
```

Reply:
Good suggestion. Done.